bench/tiered/push_pull.c

```diff
@@ -239,7 +240,17 @@ recover_validate(const char *home, uint32_t num_records, uint64_t file_size, uin
     v = (uint32_t)getpid() + num_records + (2 * counter);
     __wt_random_init_custom_seed(&rnd, v);
 
-    testutil_check(__wt_snprintf(rm_cmd, sizeof(rm_cmd), "rm -rf %s/cache-bucket/*", home));
+    if (strcmp(opts->tiered_storage_source, DIR_STORE) == 0)
+        strcpy(bucket_name, DIR_STORE_BUCKET_NAME);
+    else {
+        ret = strchr(S3_STORE_BUCKET_NAME, ';');
+        str_length = (size_t)(ret - S3_STORE_BUCKET_NAME);
+        memcpy(bucket_name, S3_STORE_BUCKET_NAME, str_length);
+        bucket_name[str_length] = '\0';
+    }
+    testutil_check(
+      __wt_snprintf(rm_cmd, sizeof(rm_cmd), "rm -rf %s/cache-%s/*", home, bucket_name));
```

Comment:
From the discussion the other day, I'd also suggest adding code to make sure there aren't any residual objects left in the database home directory. Although you set local_retention to 0, there still may be a window after it completes copying and the local object is then deleted. The test should be verifying it is gone to guarantee it is reading from the tiered storage location.

In `conn_tiered.c:tier_storage_remove_local` we add a work unit to remove the local object with a value of "now". But that object cannot be removed if the handle is open. Since the handle was open recently for writing it might be waiting for sweep. So there is a window where the object can exist locally waiting to be able to clean up.

Reply:
Good point, @sueloverso!  Perhaps we need to wait 5-10 seconds (how long?) before closing the connection after the flush_tier completes.  And then when the connection is closed, we should check the WT home directory to see if there are any *.wtobj files.  If such files were in the home dir, we could fail/stop the test, or would it be permissible to just remove those files?  This is all before opening the connection again for the "pull" side of the test.

Reply:
Makes sense.  @raviprakashgiri29, we could talk over what's required here.

Reply:
@ddanderson @sueloverso I updated the code in 5dc54e204facff54ad4cadf8cb4524b9e8dc03e7 to remove cached files and cached buckets